packages/story-editor/src/components/library/panes/text/stylePresets/stylePresets.js

```diff
@@ -0,0 +1,253 @@
+import updateProperties from '../../../../inspector/design/updateProperties';
+import getUpdatedSizeAndPosition from '../../../../../utils/getUpdatedSizeAndPosition';
+import InsertionOverlay from '../../shared/insertionOverlay';
+import { Container } from '../../../common/section';
+
+const SectionContainer = styled(Container)`
+  padding: 16px 0;
+  margin-bottom: -24px;
+  border-top: 1px solid ${({ theme }) => theme.colors.divider.tertiary};
+  border-bottom: 1px solid ${({ theme }) => theme.colors.divider.tertiary};
+`;
+
+const TitleBar = styled.div`
+  display: flex;
+  padding-bottom: 16px;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const StylesWrapper = styled.div``;
+
+const StyledButton = styled(Button)`
+  margin-left: auto;
+`;
+
+const StyledMoreButton = styled(Button)`
+  ${focusStyle};
+  margin-top: 12px;
+  padding: 0 16px;
+  justify-content: center;
+  align-self: center;
+  width: 100%;
+  svg {
+    transform: rotate(-90deg);
+    height: 32px;
+    width: 32px;
+    margin-left: -4px;
+  }
+`;
+
+const NoStylesWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: calc(100% + 32px);
+  margin: 0 0 -8px -16px;
+  background-color: ${({ theme }) => theme.colors.opacity.black24};
+  height: 64px;
+`;
+
+const NoStylesText = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+
+const SPACING = { x: 40 };
+const TYPE = 'text';
+const STYLE_BUTTON_WIDTH = 150;
+
+function PresetPanel() {
+  const isStylePaneEnabled = useFeature('libraryTextStyles');
+  const { textStyles, isText, updateSelectedElements } = useStory(
+    ({ state, actions }) => {
+      const isText =
+        state.selectedElements && areAllType(TYPE, state.selectedElements);
+      return {
+        isText,
+        textStyles: state.story.globalStoryStyles.textStyles,
+        updateSelectedElements: actions.updateSelectedElements,
+      };
+    }
+  );
+
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+
+  const buttonRef = useRef(null);
+  const stylesRef = useRef(null);
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+  const { isRTL, styleConstants: { topOffset } = {} } = useConfig();
+  const hasPresets = textStyles.length > 0;
+
+  const pushUpdate = useCallback(
+    (update) => {
+      updateSelectedElements({
+        properties: (element) => {
+          const updates = updateProperties(element, update, true);
+          const sizeUpdates = getUpdatedSizeAndPosition({
+            ...element,
+            ...updates,
+          });
+          return {
+            ...updates,
+            ...sizeUpdates,
+          };
+        },
+      });
+    },
+    [updateSelectedElements]
+  );
+
+  const handleApplyStyle = useApplyStyle({ pushUpdate });
+  const { addGlobalPreset } = useAddPreset({ presetType: PRESET_TYPES.STYLE });
```

Comment:
This is currently re-rendering due to `useAddPreset` whenever selection changes. `useAddPreset` is dependent on `selectedElements`. Might make sense to split this helper up to have separate hooks for adding a style vs colors -- perhaps this way we can only react to when selection changes from text element to non-text element and vice versa, but not when selection changes between non-text elements.

Reply:
Added issue: https://github.com/GoogleForCreators/web-stories-wp/issues/10705